### PR TITLE
Add crust-gather to e2e fail handlers

### PR DESCRIPTION
**Pull request by manno**
This replaces dump-failed-k3d.sh

* changing the job names to be fixed strings for branch protection
* updating k3s 1.30.4 for api fix
* updating k3d to support k3s 1.30.4

### CODE REVIEW

File: e2e/testenv/fail.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+func FailAndGather(message string, callerSkip ...int) {
+	if _, err := exec.LookPath("crust-gather"); err != nil {
+		ginkgo.GinkgoWriter.Print("⛔ crust-gather is not available, not dumping cluster info")
+		ginkgo.Fail(message, callerSkip...)
+	}
+
+	pwd := os.Getenv("GITHUB_WORKSPACE")
+	if pwd != "" {
+		pwd = path.Join(pwd, "tmp", "upstream")
```
Comment (weyfonk) [inline e2e/testenv/fail.go:19]: Is this why we call `crust-gather` explicitly for downstream clusters through:
```
kubectl config use-context k3d-downstream
crust-gather collect --exclude-namespace=kube-system --exclude-kind=Lease --duration=5s -f tmp/downstream
```
... but not for upstream clusters?
It looks pretty likely from the command run below, but just to be sure.

File: e2e/testenv/fail.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+	} else {
+		pwd = path.Join(os.TempDir(), "fleet-gather")
+	}
+	path := path.Join(pwd, ginkgo.CurrentSpecReport().FullText())
+
+	ginkgo.GinkgoWriter.Printf("   Gathering cluster info for '%s' to '%s'...\n", ginkgo.CurrentSpecReport().FullText(), pwd)
```
Comment (weyfonk) [inline e2e/testenv/fail.go:25]: I see squares here, presumably a fonts issue; what are these chars meant to represent? Perhaps we could use more common characters?

**Review by weyfonk — DISMISSED** [review]

File: e2e/testenv/fail.go (hunk 2 above)
Comment (manno) [reply e2e/testenv/fail.go:25]: These are little bugs...

File: e2e/testenv/fail.go (hunk 1 above)
Comment (manno) [reply e2e/testenv/fail.go:19]: I didn't want to make the fail handler downstream cluster aware. Downstreams are just dumped in the workflows for now.

**Review by weyfonk — APPROVED** [review]

**Pull request merged**